system/jlib/jtrace.cpp

```diff
@@ -107,6 +108,293 @@ class NoopSpanExporterFactory
+
+            if (span != nullptr)
+            {
+                char trace_id[32]       = {0};
+                char span_id[16]        = {0};
+                char parent_span_id[16] = {0};
+
+                span->GetTraceId().ToLowerBase16(trace_id);
+                span->GetSpanId().ToLowerBase16(span_id);
+                span->GetParentSpanId().ToLowerBase16(parent_span_id);
+
+                sout << "{"
+                << " \"Name\": \"" << span->GetName() << "\","
+                << " \"TraceId\": \"" << std::string(trace_id, 32) << "\","
+                << " \"SpanId\": \"" << std::string(span_id, 16) << "\","
+                << " \"kind\": \"" << spanKindToString(span->GetSpanKind()) << "\","
+                << " \"ParentSpanId\": \"" << std::string(parent_span_id, 16) << "\","
+                << " \"Start\": " << span->GetStartTime().time_since_epoch().count() << ","
+                << " \"Duration\": " << span->GetDuration().count() << ","
+                << " \"Attributes\": ";
+                printAttributes(span->GetAttributes());
+
+                if ( logLevel > JLogExportLevelMin)
+                {
+                    sout << ","
+                    << " \"Description\": \"" << span->GetDescription() << "\","
+                    << " \"Status\": \"" << statusMap[int(span->GetStatus())] << "\","
+                    << " \"TraceState\": \"" << span->GetSpanContext().trace_state()->ToHeader() << "\"";
+                }
+
+                if ( logLevel > JLogExportLevelDefault)
+                {
+                    sout << ", \"Events\": ";
+                        printEvents(span->GetEvents());
+                    sout << ",";
+                    sout << " \"Links\": ";
+                        printLinks(span->GetLinks());
+                    sout << ",";
+                    sout << " \"Resources\": ";
+                        printResources(span->GetResource());
+                    sout << ",";
+                    sout << " \"InstrumentedLibrary\": \"";
+                        printInstrumentationScope(span->GetInstrumentationScope());
+                    sout << "\"";
+                }
+                sout << " }";
+            }
+            LOG(MCmonitorEvent, "%s",sout.str().c_str());
+            sout.str("");
+            sout.clear();
+        }
+        return opentelemetry::sdk::common::ExportResult::kSuccess;
+    }
+
+   /**
+   * Shut down the exporter.
+   * @param timeout an optional timeout.
+   * @return return the status of the operation.
+   */
+    virtual bool Shutdown(
+      std::chrono::microseconds timeout = std::chrono::microseconds::max()) noexcept override
+    {
+        const std::lock_guard<opentelemetry::common::SpinLockMutex> locked(lock);
+        shutDown = true;
+        return true;
+    }
+
+private:
+    bool isShutDown() const noexcept
+    {
+        const std::lock_guard<opentelemetry::common::SpinLockMutex> locked(lock);
```

Comment:
Better to use std::atomic<bool> for shutdown and remove the lock.

If you did use a lock it should be the ones in jlib.